debian/apport-general-hook.py

```diff
@@ -0,0 +1,62 @@
+import json
+import logging
+
+
+def get_ec2_data(instance_data):
+    """
+    provided cloud-init data from /run/cloud/instance-data.json
+    populate a smaller set of information helpful for debugging
+    """
+    stripped_data = {}
+    instance_identity_keys = (
+        "architecture",
+        "billingProducts",
+        "imageId",
+        "region",
+    )
+    for key in instance_identity_keys:
+        stripped_data[key] = instance_data["ds"]["dynamic"][
+            "instance-identity"
+        ]["document"][key]
+    stripped_data["instanceType"] = instance_data["ds"]["meta-data"][
+        "instance-type"
+    ]
+    return stripped_data
+
+
+PLATFORM_SPECIFIC_INFO = {"ec2": get_ec2_data}
+
+
+def add_platform_specific_info(report, platform, instance_data):
+    platform_info = PLATFORM_SPECIFIC_INFO.get(platform)
+    if not platform_info:
+        return
+    retrieved_data = platform_info(instance_data)
+    for key, value in retrieved_data.items():
+        report[platform.capitalize() + key.capitalize()] = value
+
+
+def add_info(report, ui):
+    """Add a subset of non-sensitive cloud-init data"""
+    try:
+        with open("/run/cloud-init/instance-data.json", "r") as fopen:
+            instance_data = json.load(fopen)
+    except FileNotFoundError:
+        logging.getLogger().warning(
+            "cloud-init run data not found on system. "
+            "Unable to add cloud-specific data."
+        )
+        return
+
+    v1 = instance_data.get("v1")
+    if not v1:
+        logging.getLogger().warning(
+            "instance-data.json lacks 'v1' metadata. Present keys: %s",
+            sorted(instance_data.keys()),
+        )
+        return
+
+    report["CloudName"] = v1["cloud_name"]
+    report["CloudPlatform"] = v1["platform"]
+    report["CloudRegion"] = v1["region"]
```

Comment:
(separate PR) Makes me want to generalize instanceSize or instanceType from all of our datasources.
Note region can be null/None value. do we want to test for None and avoid adding it to a report when unset?

Reply:
Thanks. I'll add a check to not set them to  `None`.